### PR TITLE
Macos related updates, version in .md file fix

### PR DESCRIPTION
**Pull request by Karm**

### CODE REVIEW

File: release-scripts/mandrel-release.java [hunk 1]
```diff
@@ -476,7 +476,7 @@ private String releaseMainBody(String changelog, Set<String> jdkVersionsUsed)
             "$ curl -O -J https://code.quarkus.io/d?e=io.quarkus:quarkus-resteasy\n" +
             "$ unzip code-with-quarkus.zip\n" +
             "$ cd code-with-quarkus/\n" +
-            "$ ./mvnw package -Pnative\n" +
+            "$ ./mvnw package -Pnative -Dmaven.compiler.release=" + jdkMajorVersionExample + "\n" +
```
Comment (zakkak) [inline release-scripts/mandrel-release.java:479]: This used to not be necessary. I still never use it locally. Did anything change?

File: release-scripts/mandrel-release.java [hunk 2]
```diff
@@ -490,7 +490,9 @@ private String releaseMainBody(String changelog, Set<String> jdkVersionsUsed)
             "curl -O -J  https://code.quarkus.io/d?e=io.quarkus:quarkus-resteasy\n" +
             "unzip code-with-quarkus.zip\n" +
             "cd code-with-quarkus\n" +
-            "./mvnw package -Pnative -Dquarkus.native.container-build=true -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-mandrel-builder-image:" + version + "-java" + jdkMajorVersionExample + "\n" +
+            "./mvnw package -Pnative -Dquarkus.native.container-build=true \\\n" +
+            "    -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-mandrel-builder-image:jdk-" + jdkMajorVersionExample + " \\\n" +
+            "    -Dmaven.compiler.release=" + jdkMajorVersionExample + "\n" +
```
Comment (zakkak) [inline release-scripts/mandrel-release.java:495]: Same here.

File: release-scripts/mandrel-release.java [hunk 3]
```diff
@@ -500,9 +502,9 @@ private String releaseMainBody(String changelog, Set<String> jdkVersionsUsed)
             "powershell -c \"Invoke-WebRequest -OutFile quarkus.zip -Uri https://code.quarkus.io/d?e=io.quarkus:quarkus-resteasy\"\n" +
             "powershell -c \"Expand-Archive -Path quarkus.zip -DestinationPath . -Force\n" +
             "cd code-with-quarkus\n" +
-            "mvnw package -Pnative -Dquarkus.native.container-build=true -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-mandrel-builder-image:" + version + "-java" + jdkMajorVersionExample + "\n" +
-            "docker build -f src/main/docker/Dockerfile.native -t my-quarkus-mandrel-app .\n" +
-            "docker run -i --rm -p 8080:8080 my-quarkus-mandrel-app\n" +
+            "mvnw package -Pnative -Dquarkus.native.container-build=true -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-mandrel-builder-image:jdk-" + jdkMajorVersionExample + " -Dmaven.compiler.release=" + jdkMajorVersionExample + "\n" +
```
Comment (zakkak) [inline release-scripts/mandrel-release.java:505]: and here

**Review by zakkak — COMMENTED** [review]
LGTM, other than the addition of `-Dmaven.compiler.release=" + jdkMajorVersionExample`. Why is this now necessary while it didn't use to be?

File: release-scripts/mandrel-release.java (hunk 1 above)
Comment (Karm) [reply release-scripts/mandrel-release.java:479]: @zakkak The pom.xml says 21, so it breaks with 17. I should include a condition to append this only for JDK major 17.

File: release-scripts/mandrel-release.java (hunk 2 above)
Comment (Karm) [reply release-scripts/mandrel-release.java:495]: https://github.com/graalvm/mandrel-packaging/pull/440#discussion_r1686546138

File: release-scripts/mandrel-release.java (hunk 3 above)
Comment (Karm) [reply release-scripts/mandrel-release.java:505]: https://github.com/graalvm/mandrel-packaging/pull/440#discussion_r1686546138

File: release-scripts/mandrel-release.java (hunk 1 above)
Comment (zakkak) [reply release-scripts/mandrel-release.java:479]: Oh, so it's for Mandrel 23.0 releases only right?

In that case our instructions should probably be using Quarkus 3.2, i.e. `https://code.quarkus.io/d?S=io.quarkus.platform%3A3.2&cn=code.quarkus.io`

Some thoughts:

1. Add a quarkus-version parameter to the script
2. Edit the `23.0` branch only, but then we would need to remember to use that branch for the 23.0 releases (which we probably should anyway, but I only trust `master` t be up to date)

Comment (Karm) [reply release-scripts/mandrel-release.java:479]: @zakkak Thx. When I get to this, I will add Q version and add if-else on the version.
Let's have it all on `master`. This is the release script logic, not build logic, so having it scattered around feels like a lot of friction, far worse than an if-else branch in the code...

Comment (zakkak) [reply release-scripts/mandrel-release.java:479]: > When I get to this, I will add Q version and add if-else on the version.

@Karm can you clarify if you plan to do this as part of this PR or a different one? Thanks

Comment (Karm) [reply release-scripts/mandrel-release.java:479]: @zakkak Thx for the reminder, this PR disappeared from my radar. I am planning on refactoring this PR so as we are ready for September. Gonna amend presently...

Comment (Karm) [reply release-scripts/mandrel-release.java]: @zakkak Something like https://github.com/graalvm/mandrel-packaging/pull/440/commits/d92a3dfb241aaed7c6c39d9e0ee6684ae265a87d     ?

Comment (zakkak) [reply release-scripts/mandrel-release.java]: Yes, that looks good to me. Thanks @Karm

**Comment by Karm** [comment]
@zakkak Thx. Addressed.  Merging.

**Pull request merged**